vars/artifactSetVersion.groovy

```diff
@@ -29,17 +29,14 @@ def call(Map parameters = [:], Closure body = null) {
 
     handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
 
-        def gitUtils = parameters.juStabGitUtils ?: new GitUtils()
+        def gitUtils = parameters?.juStabGitUtils ?: new GitUtils()
+        def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
 
         if (fileExists('.git')) {
             if (sh(returnStatus: true, script: 'git diff --quiet HEAD') != 0)
                 error "[${STEP_NAME}] Files in the workspace have been changed previously - aborting ${STEP_NAME}"
         }
 
-        def script = parameters.script
```

Comment:
@daniel-kurzynski is the `artifactSetVersion` used without a `script` on your side? There is a failing test `testVersioningWithoutScript` that was provided by you in #128.

Reply:
No, we always have script. But I guess there was an issue when using
`script = [commonPipelineEnvironment: commonPipelineEnvironment]`

This is why I changed it to this:
https://github.com/SAP/jenkins-library/commit/b29de61f3c3edf375b7d2a9f93a01ab40db26a73#diff-7e959e87f33948a3000827c20ef8c7a7


The problem is that we are later calling `script.mavenExecute script: script, goals: 'versions:set'`

https://github.com/SAP/jenkins-library/blob/master/src/com/sap/piper/versioning/MavenArtifactVersioning.groovy#L16

This won't work if script is just the map with on entry commonPipelineEnvironment.

Thus, I guess in general `script = [commonPipelineEnvironment: commonPipelineEnvironment]` is not a good fallback if the script was not provided.

Maybe better `script =this`

Reply:
The test was just to ensure that the step also works without a script. But we are not using this feature.

Reply:
@OliverNocon @marcusholl @radsoulbeard @nevskrem any reason why we shouldn't use `this` as fallback for `script`? `[commonPipelineEnvironment: commonPipelineEnvironment]` anyhow uses `this.commonPipelineEnvironment`.

Reply:
Potentially cause `NotSerializableExceptions`.

Reply:
Do you have an example, when this could occur?
I would think, that every property of this usually serializable.

Reply:
@nevskrem do you know, when exactly this could occur?

Reply:
```this``` as a fallback means the pipeline step, right? Is that really intended? We have the pattern of calling the pipeline steps with e.g. ```neoDeploy script: this```. In this case the script is the pipeline script (... the ```Jenkinsfile```) and not the individual pipeline step.

Reply:
In either cases (`this` or `[common...]`) the step will *not* fail but also will **not** work as intended as the produced result (version) is not stored on an accessible `cpe`.